vendor/github.com/intel/govmm/qemu/qemu.go

```diff
@@ -986,12 +995,22 @@ func (vsock VSOCKDevice) Valid() bool {
 
 // QemuParams returns the qemu parameters built out of the VSOCK device.
 func (vsock VSOCKDevice) QemuParams(config *Config) []string {
+	var deviceParams []string
 	var qemuParams []string
 
-	deviceParam := fmt.Sprintf("%s,id=%s,%s=%d", VhostVSOCKPCI, vsock.ID, VSOCKGuestCID, vsock.ContextID)
```

Comment:
Any reason you want to switch from plain device addition to a FD based one? I thought the original code should be working already. And because the hotplug part is not FD based, I would prefer to keep the old code so that we do not complicate vsock device tracking in the runtime.

Reply:
@bergwolf  `vhostfd` help us to lock the context id, hence `vhostfd` must be inherited by QEMU process

Reply:
@devimc Thanks for the explanation. Would you please add some comments to explain the locking logic? I had to go over vhost_vsock code to understand the logic. I think it worths some texts to explain why this can work to others, e.g., vsock guest cid is registered with open vhostfd and qemu can reuse the same vhost_vsock struct we opened/registered with by passing vhostfd to it.